test/compile.jl

```diff
@@ -16,6 +16,11 @@ function redirected_stderr(expected)
     return t
 end
 
+# this environment variable would affect some error messages being tested below
+# so we disable it for the tests below
+withenv("JULIA_DEBUG_LOADING"=>"") do
+delete!(ENV, "JULIA_DEBUG_LOADING")
```

Comment:
looks like `withenv("JULIA_DEBUG_LOADING"=>nothing)` would be equivalent to this